src/backend/base/langflow/components/vectorstores/astradb.py

```diff
@@ -154,373 +166,102 @@ class AstraDBVectorStoreComponent(LCVectorStoreComponent):
+                for name, info in self.get_database_list().items()
+            ]
         except Exception as e:  # noqa: BLE001
-            self.log(f"Error fetching collections: {e}")
-
-            return ["+ Create new collection"]
-
-        return [*collections, "+ Create new collection"]
-
-    def get_collection_choice(self):
-        collection_name = self.collection_name
-        if collection_name == "+ Create new collection":
-            return self.collection_name_new
+            self.log(f"Error fetching databases: {e}")
 
-        return collection_name
+            return []
 
-    def get_collection_options(self):
-        # Only get the options if the collection exists
+    def _initialize_collection_options(self):
         database = self.get_database()
         if database is None:
-            return None
-
-        collection_name = self.get_collection_choice()
+            return []
 
         try:
-            collection = database.get_collection(collection_name, keyspace=self.keyspace or None)
-            collection_options = collection.options()
-        except Exception as _:  # noqa: BLE001
-            return None
-
-        return collection_options.vector
-
-    def update_build_config(self, build_config: dict, field_value: str, field_name: str | None = None):
-        # Always attempt to update the database list
-        if field_name in ["token", "api_endpoint", "collection_name"]:
-            # Update the database selector
-            build_config["api_endpoint"]["options"] = self._initialize_database_options()
+            collection_list = list(database.list_collections())
 
-            # Set the default API endpoint if not set
-            if build_config["api_endpoint"]["value"] == "Default database":
-                build_config["api_endpoint"]["value"] = build_config["api_endpoint"]["options"][0]
-
-            # Update the collection selector
-            build_config["collection_name"]["options"] = self._initialize_collection_options()
-
-        # Update the choice of embedding model based on collection name
-        if field_name == "collection_name":
-            # Detect if it is a new collection
-            is_new_collection = field_value == "+ Create new collection"
-
-            # Set the advanced and required fields based on the collection choice
-            build_config["embedding_choice"].update(
-                {
-                    "advanced": not is_new_collection,
-                    "value": "Embedding Model" if is_new_collection else build_config["embedding_choice"].get("value"),
-                }
-            )
-
-            # Set the advanced field for the embedding model
-            build_config["embedding_model"]["advanced"] = not is_new_collection
-
-            # Set the advanced and required fields for the new collection name
-            build_config["collection_name_new"].update(
+            return [
```

Comment:
@ogabrielluiz do you have any thoughts on the best way to support a return format like this, for a list of options?  Right now the UI is expecting a list[str]... here'd we have a list[dict] and would have a property for the display name and other pieces of data that could potentially be rendered in the dropdown. Would it be better to create a new option for DropdownInput, something like `rich_options`, or do you think we could override this to support that kind of behavior, but maintain backwards compat?

Reply:
I think the latter is better. 
Support `list[str | RichOptions]` where the dict is typed as 
```
class RichOptions(TypedDict):
	display_name: str
	properties: NotRequired[dict]
```

What do you think?

Reply:
From my perspective that definitely makes the most sense, cool!

Reply:
I think this schema is bad because it is nested. 
Maybe we could just have a dict that must contain display_name and all other keys are used by the component itself. 
We could add id as well.